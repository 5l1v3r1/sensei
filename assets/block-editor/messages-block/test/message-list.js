/**
 * External dependencies.
 */
import renderer from 'react-test-renderer';

/**
 * Internal dependencies.
 */
import MessageList from '../message-list';

const messages = [
	{
		id: 100,
		link: 'http://example.com/messages/100',
		displayed_title: 'Re: My Course',
		sender: 'user1',
		displayed_date: '2019-11-16',
		excerpt: {
			rendered: 'Hello, teacher!',
		},
	},
	{
		id: 101,
		link: 'http://example.com/messages/101',
		displayed_title: 'Re: My Lesson',
		sender: 'user2',
		displayed_date: '2019-11-17',
		excerpt: {
			rendered: 'Hello, lesson teacher!',
		},
	},
];

describe( 'MessageList Component', () => {
	it( 'renders the given messages', () => {
		const tree = renderer
			.create( <MessageList messages={ messages } /> )
			.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( 'renders the placeholder UI when fetching', () => {
		const tree = renderer
			.create( <MessageList isFetching={ true } messages={ messages } /> )
			.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
} );
