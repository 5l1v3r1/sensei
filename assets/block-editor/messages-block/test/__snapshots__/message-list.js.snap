// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList Component renders the given messages 1`] = `
<div
  className="message-container"
>
  <article>
    <a
      className="message-link"
      href="http://example.com/messages/100"
    >
      <h3>
        Re: My Course
      </h3>
    </a>
    <p
      className="message-meta"
    >
      <small>
        <em>
          Sent by user1 on 2019-11-16
        </em>
      </small>
    </p>
    <section
      className="entry"
    >
      <p
        className="message-excerpt"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Hello, teacher!",
          }
        }
      />
    </section>
  </article>
  <article>
    <a
      className="message-link"
      href="http://example.com/messages/101"
    >
      <h3>
        Re: My Lesson
      </h3>
    </a>
    <p
      className="message-meta"
    >
      <small>
        <em>
          Sent by user2 on 2019-11-17
        </em>
      </small>
    </p>
    <section
      className="entry"
    >
      <p
        className="message-excerpt"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Hello, lesson teacher!",
          }
        }
      />
    </section>
  </article>
</div>
`;

exports[`MessageList Component renders the placeholder UI when fetching 1`] = `
<div
  className="message-container is-fetching"
>
  <article>
    <a
      className="message-link"
    >
      <h3 />
    </a>
    <p
      className="message-meta"
    >
      <small>
        <em>
           
        </em>
      </small>
    </p>
    <section
      className="entry"
    >
      <p
        className="message-excerpt"
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
    </section>
  </article>
  <article>
    <a
      className="message-link"
    >
      <h3 />
    </a>
    <p
      className="message-meta"
    >
      <small>
        <em>
           
        </em>
      </small>
    </p>
    <section
      className="entry"
    >
      <p
        className="message-excerpt"
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
    </section>
  </article>
</div>
`;
